src/result/result.py

```diff
@@ -336,6 +431,314 @@ def or_else(self, op: Callable[[E], Result[T, F]]) -> Result[T, F]:
+        already an Error, so just returns self.
+
+        Example:
+            assert Err(2) @ (lambda x:x**2) == Err(2)
+        """
+        return self
+
+    async def __ge__(self, op: Callable[[T], Awaitable[U]]) -> Result[U, E]:
+        """
+        Piping for async functions.
+        Just returns self, but needs to be awaited to have the same behavoiur as the Ok variant.
+        The operator CANNOT chain without await
+
+        Example:
+            async def foo(x): return x+1
+            x = Err('foo') >= foo
+            result = await x
+            assert result == Err('foo')
+
+            ## can be inlined with parentheses, but make sure to LEAVE SPECE AFTER AWAIT
+            y = await (await ( Err('bar') >= foo ) @ (lambda x: x+1) >= foo)
+            assert y == Err('bar')
+
+        """
+        return self
+
+    def __mod__(self, op: Callable[[T], bool]) -> Result[T, E]:
+        """
+        pipe filter.
+        filtered values are treaded as errors
+        returns self, since it's already an error
+        """
+        return self
+
+    def __or__(self, other: Result[Any, Any]) -> MultiResult:
+        """
+        Joining operator.
+        If not specified manually, the Result needs to initialize MultiResult instance.
+
+        Example:
+            mres: MultiResult = Ok(2) | Ok(12)
+            assert tuple(mres.results) == tuple([ Ok(2) , Ok(12) ])
+        """
+        return MultiResult(self, other)
+
+
+class MultiResult:
+    """
+    A Result class for storing (joining) multiple results to use in multivariate functions.
+    The multivariate operator has the lower priority so it gets evaluated AFTER:
+        - single argument functions ( Result @ fnc -> Result )
+        - result joining ( Result | Result -> MultiResult )
+        - finally then: ( MultiResult > multivarfnc -> Result )
+        - similarly for async: ( MultiResult >= async multifnc -> coroutine )
+            ... this needs to be awaited
+    """
+
+    __match_args__ = ("results",)
+    __slots__ = ("results",)
+
+    def __init__(self, *results: Result[Any, Any]) -> None:
+        self.results = list(results)
+
+    def __repr__(self) -> str:
+        results = [str(res) for res in self.results]
+        return "MultiResult(" + ",".join(results) + ")"
+
+    def __eq__(self, other: Any) -> bool:
+        return (
+            isinstance(other, MultiResult)
+            and (n := len(self.results)) == len(other.results)
+            and all([self.results[i] == other.results[i] for i in range(n)])
+        )
+
+    def __ne__(self, other: Any) -> bool:
+        return not (self == other)
+
+    def __hash__(self) -> int:
+        """
+        hash by repr if normal fails
+        """
+        try:
+            return hash(tuple(hash(res) for res in self.results))
+        except Exception:
+            return hash(repr(self))
+
+    def is_ok(self) -> bool:
+        return all([res.is_ok() for res in self.results])
```

Comment:
no need to construct a list here, `all()` takes an iterable which includes generator expressions, e.g. `all(x for x in ...)`

Reply:
Fixed. Thanks for the tip!